executor/aggfuncs/aggfuncs.go

```diff
@@ -177,6 +180,18 @@ type AggFunc interface {
 	// partial result and then calculates the final result and append that
 	// final result to the chunk provided.
 	AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error
+
+	// NewSpillSerializeHelper creates a spillSerializeHelper.
+	NewSpillSerializeHelper() *SpillSerializeHelper
+
+	// SerializeToChunk will serialize meta data of aggregate function into bytes and put them into chunk.
+	SerializeForSpill(_ sessionctx.Context, partialResult PartialResult, chk *chunk.Chunk, spillHelper *SpillSerializeHelper)
+
+	// DeserializeToPartialResultForSpill deserializes from bytes to PartialResult.
+	DeserializeToPartialResultForSpill(sctx sessionctx.Context, src *chunk.Chunk) ([]PartialResult, int64, error)
+
+	// deserializeForSpill deserialize from bytes to aggregate function.
+	deserializeForSpill(helper *spillDeserializeHelper) (PartialResult, int64, error)
```

Comment:
we can extract this part into an individual interface like 
``` go
type aggSerializeHelper interface {
	// SerializeToChunk will serialize meta data of aggregate function into bytes and put them into chunk.
	Serialize(_ sessionctx.Context, partialResult PartialResult, chk *chunk.Chunk, spillHelper *SpillSerializeHelper)

	// DeserializeToPartialResultForSpill deserializes from bytes to PartialResult.
	Deserialize(sctx sessionctx.Context, src *chunk.Chunk) ([]PartialResult, int64, error)
}
```

Reply:
``` go
type AggFunc interface {
	aggSerializeHelper
    ...
}
```

Reply:
> we can extract this part into an individual interface like
> 
> ```go
> type aggSerializeHelper interface {
> 	// SerializeToChunk will serialize meta data of aggregate function into bytes and put them into chunk.
> 	Serialize(_ sessionctx.Context, partialResult PartialResult, chk *chunk.Chunk, spillHelper *SpillSerializeHelper)
> 
> 	// DeserializeToPartialResultForSpill deserializes from bytes to PartialResult.
> 	Deserialize(sctx sessionctx.Context, src *chunk.Chunk) ([]PartialResult, int64, error)
> }
> ```

done